mint-auditor/api/proto/mint_auditor.proto

```diff
@@ -19,6 +20,73 @@ service MintAuditorApi {
     rpc GetCounters(google.protobuf.Empty) returns (Counters) {}
 }
 
+/// A Gnosis Safe transaction.
+message GnosisSafeTx {
+    /// The Ethererum transaction hash.
+    string eth_tx_hash = 1;
+
+    /// The raw JSON transaction, as received from the Gnosis API service.
+    string raw_tx_json = 2;
+}
+
+/// A Gnosis Safe deposit (transfer into the safe).
+message GnosisSafeDeposit {
+    /// Database id.
+    int32 id = 1;
+
+    /// Ethereum transaction hash.
+    string eth_tx_hash = 2;
+
+    /// Ethereum block number.
+    uint64 eth_block_number = 3;
+
+    /// Safe address.
+    string safe_addr = 4;
+
+    /// The token address.
+    string token_addr = 5;
+
+    /// The amount being transferred.
+    uint64 amount = 6;
+
+    /// Expected nonce of the matching MintTx.
+    string expected_mc_mint_tx_nonce_hex = 7;
+}
+
+/// A Gnosis Safe withdrawal (transfer out of the safe).
+message GnosisSafeWithdrawal {
+    /// Database id.
+    int32 id = 1;
+
+    /// Ethereum transaction hash.
+    string eth_tx_hash = 2;
+
+    /// Ethereum block number.
+    uint64 eth_block_number = 4;
+
+    /// Safe address
+    string safe_addr = 5;
+
+    /// Token address.
+    string token_addr = 6;
+
+    /// The amount being transferred.
+    uint64 amount = 7;
+
+    /// Associated MobileCoin TxOut public key.
+    external.CompressedRistretto mc_tx_out_pub_key = 8;
+}
+
+/// Audit data for a single mint transaction.
+message AuditedMintTx {
+    /// The actual MintTx.
+    external.MintTx mint_tx = 1;
+
+    /// Matching Gnosis Safe deposit.
+    GnosisSafeDeposit gnosis_safe_deposit = 2;
+}
+
+/// Audit data for a single block.
```

Comment:
I was originally planning on including API endpoints for exposing this data in this PR, but it ended up being large enough that I opted to postpone this for a future PR.